orchagent/portsorch.cpp

```diff
@@ -4631,3 +4651,97 @@ bool PortsOrch::initGearboxPort(Port &port)
     return true;
 }
 
+/*
+ * The following method is called from setVoqInbandIntf() introduced in the following PR:
+ * https://github.com/Azure/sonic-swss/pull/1431
+ */
+bool PortsOrch::addInbandVlan(const string &alias, const string &type)
+{
+    if (m_inbandPortName == alias)
+    {
+        /* Inband vlan already exists with this name */
+        SWSS_LOG_WARN("Interface %s is already configured as inband!", alias.c_str());
+        return true;
+    }
+
+    if (type != "vlan")
+    {
+        SWSS_LOG_WARN("Inband type %s is not vlan.", type.c_str());
+        return false;
+    }
+    
+    /* Create inband Vlan and also local and remote CPU ports to inband Vlan */
+
+    Port cpuPort;
+    Port inbandVlan;
+    string tagged_mode("tagged");
+
+    if (!addVlan(alias))
+    {
+        SWSS_LOG_ERROR("Failed to add inband VLAN %s", alias.c_str());
+        return false;
+    }
+
+    if (!getPort(alias, inbandVlan))
+    {
+        SWSS_LOG_ERROR("Failed to get port object of inband VLAN %s", alias.c_str());
+        return false;
+    }
```

Comment:
Can we not do this vlan addition processing via vlan configuration by having entry in VLAN table in config_db? So that this will be similar to inband port type. For port type, the VOQ_INBAND_INTERFACE configuration is just to designate a port or vlan for interface for cpu-to-cpu communication. We actually do not configure the port or vlan itself.

Reply:
We thought about this too. But inband VLAN is not exactly the same as regular SONiC VLANs. For example, the inband VLAN members are cpu ports, which do not have host intfs being created today.  Therefore, inband VLAN processing is different from VLAN processing.